.github/workflows/build-cache.yml

```diff
@@ -29,13 +29,24 @@ jobs:
       run: cargo build --all-targets --all --all-features --profile pr-tests
     - name: Check without Halo2
       run: cargo check --all --no-default-features --profile pr-tests
+    - name: Delete previous cache
+      uses: WarpBuilds/cache@v1
+      with:
+        path: |
+          ~/.cargo/registry/index/
+          ~/.cargo/registry/cache/
+          ~/.cargo/git/db/
+          target/
+          Cargo.lock
+        key: ${{ runner.os }}-cargo-pr-tests
+        delete-cache: true
```

Comment:
what does this delete-cache do?

Reply:
idea is to keep a single cache entry?

Reply:
Yes. This cache is different from GitHub/BuildJet in two ways: costs money to keep, and you can explicitly delete it. Since we don't need more than one, we can do like this instead of using the hash trick.

Reply:
What if the build fails? Does it keep the old cache?

Reply:
maybe we can do sth like add the "date" to the cache name and then delete after saving?
also, do we need the `path` entry here for deleting the cache? just the key should be fine no?

Reply:
The problem is that for deleting, we need an exact match, so it is less riskier of leaking stale cache, this way, using just one name.

About the using the path, it is in their example. I just followed.

Reply:
I mean if this only runs on main, the chances of the build failing are rather low.

Reply:
Yeah, but even then, deletion only happens if the previous steps complete successfully. There is the minimal chance delete completes but the instance crashes before it can complete the upload, but then you can just run the job manually.